### PR TITLE
[WIP] Changes to work with DeepChem 2.4

### LINKED ISSUE #26: Updating Book Code to DeepChem 2.4.0
We need to work to update the book code to DeepChem 2.4.0. I've gotten a report on gitter than the chapter 6 code isn't working. What else do we need to update?

CC @peastman

### PR DESCRIPTION
**Pull request by peastman**
This fixes #26.

There are a few issues still remaining.  The examples that use RdkitGridFeaturizer don't work correctly.  We need to fix it in DeepChem.  There also are a couple of examples that run but produce worse results than they used to: the DR and RNAi examples.  (That's separate from the ones we *expect* to give different results, due to moving from random splitting to scaffold splitting.)

### CODE REVIEW

File: Chapter05/pdbbind_nn.py [hunk 1]
```diff
@@ -1,7 +1,9 @@
 # Use a neural network to predict the PDBBind dataset.  First load the data.
 
 import deepchem as dc
-pdbbind_tasks, pdbbind_datasets, transformers = dc.molnet.load_pdbbind(featurizer="grid", split="random", subset="core")
+featurizer = dc.feat.RdkitGridFeaturizer(voxel_width=2.0, sanitize=True, flatten=True,
```
Comment (rbharath) [inline Chapter05/pdbbind_nn.py:4]: Will start doing some work on `RdkitGridFeaturizer` this week to get the codebase to a more sensible spot and fix our issues...

**Review by rbharath — COMMENTED** [review]
Did a first pass and looks good! I'll plan to work on `RdkitGridFeaturizer` this week

**Comment by rbharath** [comment]
@peastman Can we go ahead and merge this PR in?

**Comment by peastman** [comment]
Sure, I'll merge it.  As noted above, there are still some problems though.  We'll need to fix them separately.

**Pull request merged**